src/V3WidthSel.cpp

```diff
@@ -215,6 +215,18 @@ class WidthSelVisitor final : public VNVisitor {
         }
     }
 
+    static bool isPossibleWrite(AstNodeExpr* nodep) {
+        AstNode* abovep = nodep->firstAbovep();
+        if (AstNodeAssign* const assignp = VN_CAST(abovep, NodeAssign)) {
+            return assignp->lhsp() == nodep;
+        }
+        if (AstMethodCall* const methodCallp = VN_CAST(abovep, MethodCall)) {
+            return methodCallp->fromp() == nodep;
+        }
+        AstNodeExpr* exprp = VN_CAST(abovep, NodeExpr);
+        return exprp ? isPossibleWrite(exprp) : false;
+    }
+
```

Comment:
This seems sensitive to how the code is written, e.g. an assignment of a queue of queues wouldn't work.  Maybe we need a different AstNode for the writes?

Reply:
We still need to recognize the need for such a node. There's no separate grammar or anything for it. Unless I'm missing something.

I found a case that I forgot to consider, which is `q[p[0]] = x` appending to `p`. Pushed a test and fixed it. Of course, there may be more bugs like this, but I don't see a better approach for now.

I'm not sure what you mean with the queue of queues example, it seems to work (see latest push).

Reply:
If the test works, I'm satisfied.